apps/dashboard/src/components/primitives/field-editor/variable-popover/variable-popover.tsx

```diff
@@ -0,0 +1,257 @@
+import { useState, useRef, useEffect } from 'react';
+import { PopoverContent } from '@/components/primitives/popover';
+import { Input, InputField } from '@/components/primitives/input';
+import { FormControl, FormItem } from '@/components/primitives/form/form';
+import { Switch } from '@/components/primitives/switch';
+import { Code2 } from '../../../icons/code-2';
+import { Separator } from '../../separator';
+import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '@/components/primitives/select';
+import { TransformerItem } from './components/transformer-item';
+import { TransformerList } from './components/transformer-list';
+import { useVariableParser } from './hooks/use-variable-parser';
+import { useTransformerManager } from './hooks/use-transformer-manager';
+import { formatLiquidVariable } from './utils';
+import type { VariablePopoverProps } from './types';
+
+export function VariablePopover({ variable, onClose, onUpdate }: VariablePopoverProps) {
```

Comment:
![Screenshot 2024-12-31 at 10 03 20](https://github.com/user-attachments/assets/10d97e21-6cf0-402b-8912-761c9dce11ed)

Reply:
when the body has validation errors they are shown in the popover fields

https://github.com/user-attachments/assets/e657edc3-3132-44c7-a65c-53e997515848

Reply:
the borders and headers of two seems to be a bit different

![Screenshot 2024-12-31 at 10 05 40](https://github.com/user-attachments/assets/bae7f0e0-64d6-4e5d-ab61-bd9e95bcdab3)